packages/apps/marmalade-marketplace/src/components/Sale/RegularSale.tsx

```diff
@@ -0,0 +1,112 @@
+import { useEffect } from "react";
+import { buyToken, getEscrowAccount } from "@kadena/client-utils/marmalade";
+import { getWebauthnGuard } from "@kadena/client-utils/webauthn";
+import * as styles from "@/styles/sale.css"
+import { env } from "@/utils/env";
+import { Button } from "@kadena/kode-ui";
+import { createSignWithSpireKey } from "@/utils/signWithSpireKey";
+import { Sale } from "@/hooks/getSales";
+import { useRouter, useSearchParams } from "next/navigation";
+import { AppRouterInstance } from "next/dist/shared/lib/app-router-context.shared-runtime";
+import { useAccount } from "@/hooks/account";
+import { PactNumber } from "@kadena/pactjs";
+import { generateSpireKeyGasCapability } from "@/utils/helper";
+
+export interface RegularSaleProps {
+  tokenImageUrl: string;
+  sale: Sale
+}
+
+export function RegularSale({ tokenImageUrl, sale }: RegularSaleProps) {
+
+  const router = useRouter() as AppRouterInstance;
+  const searchParams = useSearchParams();
+  const { account } = useAccount();
+
+  useEffect(() => {
+
+    const transaction = searchParams.get("transaction");
```

Comment:
In what case do we get here with a transaction in the queryparams?

Reply:
When the signed transaction from the spirekey is returned the page is directed to the `/transaction`page.

Reply:
As discussed it would be better if spirekey was redirected to the transaction page directly, which would remove the need for this extra redirect.
However since this call is being made from client-utils we need to update that first with the ability to specify the return url.